includes/class-newspack-newsletters-subscription.php

```diff
@@ -301,5 +302,30 @@ public static function add_contact( $contact, $lists = [] ) {
 
 		return $result;
 	}
+
+	/**
+	 * Handle Newspack's reader registration – add contact to list.
+	 *
+	 * @param string         $email         Email address.
+	 * @param bool           $authenticate  Whether to authenticate after registering.
+	 * @param false|int      $user_id       The created user id.
+	 * @param false|\WP_User $existing_user The existing user object.
+	 * @param false|any      $metadata Metadata.
+	 */
+	public static function newspack_registered_reader( $email, $authenticate, $user_id, $existing_user, $metadata ) {
+		if ( $existing_user ) {
+			// Only sign up if they've just created an account.
+			return;
+		}
```

Comment:
We shouldn't mind if it's an existing user. Reader registration is optimistic and the subscription is upserted to that email, so we would want that to happen.

Reply:
Done in 9bedd8aa04daf81c30156878b0f97ae9a11746bb. So it looks like the `add_contact` method should be called `upsert_contact`.